File: src/examples/NotionGraphApi/Interface/Analysis/QuerySpecification.cs
namespace NotionGraphApi.Interface.Analysis;

public class QuerySpecification
{
    public List<QueryDatabaseReference> DatabaseReferences { get; } = new();
    public List<QueryDatabaseReference> AllPropertiesSelected { get; } = new();
    public List<OutputPropertySelection> PropertiesSelected { get; } = new();
    public List<SelectStep> SelectSteps { get; } = new();
}

File: frontend/browser/src/api/notion-api/model/analysis/QuerySpecification.ts
import QueryDatabaseReference from "./QueryDatabaseReference.ts";
import OutputPropertySelection from "./OutputPropertySelection.ts";
import QuerySelectStep from "./QuerySelectStep.ts";

export default interface QuerySpecification {
    databaseReferences : QueryDatabaseReference[],
    allPropertiesSelected: QueryDatabaseReference[],
    propertiesSelected: OutputPropertySelection[],
    selectSteps: QuerySelectStep[]
}